File: BukkitUtil-core/src/main/java/no/kh498/util/jackson/mixin/ConfigurationSerializableMixIn.java
package no.kh498.util.jackson.mixin;

import com.fasterxml.jackson.annotation.JsonValue;

import java.util.Map;

/**
 * @author Elg
 */
public interface ConfigurationSerializableMixIn {

    @JsonValue
    Map<String, Object> serialize();
}

File: BukkitUtil-core/src/main/kotlin/no/kh498/util/jackson/mixIn/ConfigurationSerializableMixIn.kt
package no.kh498.util.jackson.mixIn

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonTypeInfo
import org.bukkit.configuration.serialization.ConfigurationSerialization

/**
 * @author Elg
 */
@JsonInclude(JsonInclude.Include.NON_DEFAULT)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = ConfigurationSerialization.SERIALIZED_TYPE_KEY)
interface ConfigurationSerializableMixIn {


}
